### PR TITLE
fix for Synology devices

### LINKED ISSUE #52: ZFS can cause the container to fail on setfacl
This isn't something that is strictly the fault of docker-postgresql, but maybe it should have a mention somewhere?  Or you could test for setfacl capability before trying to run the command?  I'm not sure, but I thought I'd record it here.

The situation is that when using the zfs docker filesystem driver, it uses cloning and snapshots from zfs natively.  ZFS (on Linux, at least) has the capability to not use POSIX ACLs.  This causes a setfacl on /etc/resolv.conf to fail in the container, which makes the container fail.

The fix is to set "acltype" to "posixacl" on the underlying zfs filesystem (and restart the container).  This allows setfacl to work.

Thanks,
-Eric

### PR DESCRIPTION
**Pull request by gopeter**
This MR fixes #52 (problem appears since #42)

I just created a [new image on hub.docker.com](https://hub.docker.com/r/debd/docker-postgresql/) and everything works fine now on my Synology :)

### CODE REVIEW

File: runtime/functions [hunk 1]
```diff
@@ -260,7 +260,11 @@ trust_localnet() {
 
 set_resolvconf_perms() {
   echo "Setting resolv ACLs..."
-  setfacl -m user:${PG_USER}:r /etc/resolv.conf
+  if which setfacel >/dev/null; then
```
Comment (sameersbn) [inline runtime/functions]: typo?

File: runtime/functions [hunk 2]
```diff
@@ -260,7 +260,11 @@ trust_localnet() {
...
+    setfacl -m user:${PG_USER}:r /etc/resolv.conf
```
Comment (sameersbn) [inline runtime/functions]: maybe just this should suffice:

```bash
setfacl -m user:${PG_USER}:r /etc/resolv.conf || true
```

Comment (gopeter) [inline runtime/functions]: I'll try this out and update my PR today. Thanks :)

Comment (gopeter) [inline runtime/functions]: Nope, this doesn't work on my Synology: 
````
$ setfacl || true
$ setfacl: not found
````

File: runtime/functions (hunk 1 above)
Comment (gopeter) [inline runtime/functions]: You're right. I'll try some other things

**Comment by gopeter** [comment]
The new check works. I've tried it on my Synology, where the command fails, and on my MacBook, where the ACLs were set correctly.

File: runtime/functions (hunk 2 above)
Comment (sameersbn) [inline runtime/functions]: This cannot be true. `setfacl` is installed in the image, its a part of the image

```bash
# docker run -it --rm sameersbn/postgresql setfacl
Usage: setfacl [-bkndRLP] { -m|-M|-x|-X ... } file ...
Try `setfacl --help' for more information.
```
Is it possible your using an old version of the image?

File: runtime/functions [hunk 3]
```diff
@@ -259,8 +259,11 @@ trust_localnet() {
...
 
 set_resolvconf_perms() {
-  echo "Setting resolv ACLs..."
-  setfacl -m user:${PG_USER}:r /etc/resolv.conf
+  if setfacl -m user:${PG_USER}:r /etc/resolv.conf; then
+    echo "Setting resolv.conf ACLs..."  
+  else
+    echo "resolv.conf ACLs couldn't be set: `setfacl` not supported"
+  fi
 }
```
Comment (sameersbn) [inline runtime/functions]: I'd prefer 

```bash
echo "Setting resolv.conf ACLs..."
setfacl -m user:${PG_USER}:r /etc/resolv.conf || true
```

I cannot understand why you would see this error inside the container.

> ```
> $ setfacl || true
> $ setfacl: not found
> ```

Comment (gopeter) [inline runtime/functions]: OK, one moment, I'll give it another try!

**Comment by gopeter** [comment]
OK, you're absolutely right, your syntax works too. Thanks you very much!

**Comment by sameersbn** [comment]
@gopeter thanks for taking the time to resolve the issue.

**Pull request merged**